### PR TITLE
Adding Vesuvio analysis

### LINKED ISSUE #29040: Adding new vesuvio analysis algorithm
One of the scientists has provided an example script for the Vesuvio workflow. This issue is to add it to mantid.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
This adds a new method for doing vesuvio analysis. This method is still in development by the instrument scientists, hence the tests are lightweight (the analysis will probably change).

**To test:**

Open Mantid and run the script from the doc. Check that it runs ok (it can be slow). 

Fixes #29040. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/VesuvioAnalysis.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+The Y-space fit is performed using the Mantid minimiser and average Mantid resolution function, using
+a Gauss-Hermite expansion including H0 and H4 at present, while H3 (proportional to final-state effects)
+is not needed as a result of the symmetrisation.
+'''
+
+
+class VesuvioAnalysis(PythonAlgorithm):
+    def category(self):
+        return "Indirect"
```
Comment (AnthonyLim23) [inline Framework/PythonInterface/plugins/algorithms/VesuvioAnalysis.py:66]:
```suggestion
        return "Inelastic\\Indirect\\Vesuvio"
```

File: docs/source/algorithms/VesuvioAnalysis-v1.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+.. testoutput::
+
+   variable value
+   f1.sigma1 4.917
+   f1.c4 0.000
+   f1.c6 0.000
+   f1.A 0.080
+   f1.B0 0.000
+   Cost function value 0.203
```
Comment (AnthonyLim23) [inline docs/source/algorithms/VesuvioAnalysis-v1.rst:78]:
```suggestion
   Cost function value 0.200
```

File: docs/source/algorithms/VesuvioAnalysis-v1.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+   print("variable", "value")
+   for row in range(fit_results.rowCount()):
+       print(fit_results.column(0)[row],"{:.3f}".format(fit_results.column(1)[row]))
+   
+Output:
+
+.. testoutput::
+
+   variable value
+   f1.sigma1 4.917
```
Comment (AnthonyLim23) [inline docs/source/algorithms/VesuvioAnalysis-v1.rst:73]:
```suggestion
   f1.sigma1 4.931
```

File: Framework/PythonInterface/plugins/algorithms/VesuvioAnalysis.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (Pasarus) [inline Framework/PythonInterface/plugins/algorithms/VesuvioAnalysis.py:3]:
```suggestion
# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
```

File: Framework/PythonInterface/test/python/plugins/algorithms/VesuvioAnalysisTest.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (Pasarus) [inline Framework/PythonInterface/test/python/plugins/algorithms/VesuvioAnalysisTest.py:3]:
```suggestion
# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
```

File: docs/source/release/v5.1.0/indirect_geometry.rst [hunk 6]
```diff
@@ -52,4 +52,16 @@ Bug Fixes
...
+New
+###
+
+- :ref:`VesuvioAnalysis <algm-VesuvioAnalysis>` has been added as a beta method for analysing Vesuvio data.
+
+
+New
+###
+
+- :ref:`VesuvioAnalysis <algm-VesuvioAnalysis>` has been added as a beta method for analysing Vesuvio data.
```
Comment (Pasarus) [inline docs/source/release/v5.1.0/indirect_geometry.rst:64]: This seems to have been added twice to the release notes, please remove one.

File: scripts/Inelastic/vesuvio/analysisHelpers.py [hunk 7]
```diff
@@ -0,0 +1,582 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (Pasarus) [inline scripts/Inelastic/vesuvio/analysisHelpers.py:3]:
```suggestion
# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
```

**Review by Pasarus — CHANGES_REQUESTED** [review]
I have resisted the urge to attempt to correct some programming issues in the provided scripts (As they are scientist experimental scripts), just need to fix the copyright in those files and remove the duplicate release notes :+1:

File: docs/source/release/v5.1.0/indirect_geometry.rst [hunk 8]
```diff
@@ -52,4 +52,16 @@ Bug Fixes
...
 - Indirect Data Analysis F(Q) fit multiple workspaces can now load more than one spectra from each workspace.
 
+New
+###
+
+- :ref:`VesuvioAnalysis <algm-VesuvioAnalysis>` has been added as a beta method for analysing Vesuvio data.
+
+
+New
+###
```
Comment (AnthonyLim23) [inline docs/source/release/v5.1.0/indirect_geometry.rst:62]:
```suggestion
```

File: docs/source/release/v5.1.0/indirect_geometry.rst [hunk 9]
```diff
@@ -52,4 +52,16 @@ Bug Fixes
...
+
```
Comment (AnthonyLim23) [inline docs/source/release/v5.1.0/indirect_geometry.rst:63]:
```suggestion
```

File: docs/source/release/v5.1.0/indirect_geometry.rst (hunk 6 above)
Comment (AnthonyLim23) [inline docs/source/release/v5.1.0/indirect_geometry.rst:64]:
```suggestion
```

Comment (AnthonyLim23) [reply docs/source/release/v5.1.0/indirect_geometry.rst:64]: not sure how that happened

File: docs/source/release/v5.1.0/indirect_geometry.rst [hunk 10]
```diff
@@ -52,4 +52,16 @@ Bug Fixes
...
+- :ref:`VesuvioAnalysis <algm-VesuvioAnalysis>` has been added as a beta method for analysing Vesuvio data.
+
```
Comment (AnthonyLim23) [inline docs/source/release/v5.1.0/indirect_geometry.rst:65]:
```suggestion
```

**Review by Pasarus — APPROVED** [review]
Looks as Good as can be with experimental scientist code :+1:

**Pull request merged**